File: Programmers/Level1/체육복.py
def solution(n, lost, reserve):
    count = 0
    canparticipate = [False]*len(lost)
    for i in range(len(lost)):
        for j in reserve:
            if lost[i] == j:
                canparticipate[i] = True
                reserve.remove(j)
                break

    for i in range(len(lost)):
        for j in reserve:
            if(canparticipate[i]):
                continue
            if(lost[i] - 1 == j or lost[i]+1 == j):
                canparticipate[i] = True
                reserve.remove(j)
                break

    for b in canparticipate:
        if(b == False):
            count += 1

    return n - count

File: Swift 코딩테스트 연습/체육복.swift
import Foundation

func solution(_ n:Int, _ lost:[Int], _ reserve:[Int]) -> Int {
    var clothStatus: [Int] = Array(repeating: 1, count: n)
    var answer: Int = 0
    
    for i in 0..<lost.count {
        clothStatus[lost[i] - 1] -= 1
    } 
    
    for i in 0..<reserve.count {
        clothStatus[reserve[i] - 1] += 1
    }

    for i in 0..<n {
        if clothStatus[i] == 0 {
            if i >= 1 && clothStatus[i-1] > 1 {
                clothStatus[i-1] -= 1
                clothStatus[i] += 1
            } else if i < n - 1 && clothStatus[i+1] > 1 {
                clothStatus[i+1] -= 1
                clothStatus[i] += 1
            } 
        }
    }

    for i in 0..<n {
        if clothStatus[i] != 0 {
            answer += 1
        }
    }
    
    return answer
}